File: common/src/main/java/dev/number6/db/model/converter/SentimentScoreConverter.java
package dev.number6.db.model.converter;

import dev.number6.db.model.SentimentScore;
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Collection;

public class SentimentScoreConverter implements DynamoDBTypeConverter<String, Collection<SentimentScore>> {

    private final Gson gson = new Gson();

    @Override
    public String convert(Collection<SentimentScore> scores) {
        return gson.toJson(scores);
    }

    @Override
    public Collection<SentimentScore> unconvert(String object) {
        Type type = new TypeToken<Collection<SentimentScore>>() {
        }.getType();
        return gson.fromJson(object, type);
    }
}

File: common/src/main/java/dev/number6/db/model/converter/SentimentScoreConverter.kt
package dev.number6.db.model.converter

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import dev.number6.db.model.SentimentScore

class SentimentScoreConverter : DynamoDBTypeConverter<String, Collection<SentimentScore>> {
    private val gson = Gson()
    override fun convert(scores: Collection<SentimentScore>): String {
        return gson.toJson(scores)
    }

    override fun unconvert(`object`: String): Collection<SentimentScore> {
        val type = object : TypeToken<Collection<SentimentScore>>() {}.type
        return gson.fromJson(`object`, type)
    }
}